packages/dd-trace/test/datastreams/pathway.spec.js

```diff
@@ -40,4 +47,72 @@ describe('encoding', () => {
     expect(decoded.pathwayStartNs).to.equal(expectedContext.pathwayStartNs)
     expect(decoded.edgeStartNs).to.equal(expectedContext.edgeStartNs)
   })
+  it('should encode and decode to the same value when using base64', () => {
+    const ctx = {
+      pathwayStartNs: 1685673482722000000,
+      edgeStartNs: 1685673506404000000
+    }
+    ctx.hash = computePathwayHash('test-service', 'test-env',
+      ['direction:in', 'group:group1', 'topic:topic1', 'type:kafka'], Buffer.from('0000000000000000', 'hex'))
+
+    const encodedPathway = encodePathwayContextBase64(ctx)
+    const decodedPathway = decodePathwayContextBase64(encodedPathway)
+
+    expect(decodedPathway.hash.toString()).to.equal(ctx.hash.toString())
+    expect(decodedPathway.pathwayStartNs).to.equal(ctx.pathwayStartNs)
+    expect(decodedPathway.edgeStartNs).to.equal(ctx.edgeStartNs)
+  })
+
+  it('should encode and decode to the same value when using the PathwayCodec', () => {
+    const ctx = {
+      pathwayStartNs: 1685673482722000000,
+      edgeStartNs: 1685673506404000000
+    }
+    const carrier = {}
+    ctx.hash = computePathwayHash('test-service', 'test-env',
+      ['direction:in', 'group:group1', 'topic:topic1', 'type:kafka'], Buffer.from('0000000000000000', 'hex'))
+
+    DsmPathwayCodec.encode(ctx, carrier)
+    const decodedCtx = DsmPathwayCodec.decode(carrier)
+
+    expect(decodedCtx.hash.toString()).to.equal(ctx.hash.toString())
+    expect(decodedCtx.pathwayStartNs).to.equal(ctx.pathwayStartNs)
+    expect(decodedCtx.edgeStartNs).to.equal(ctx.edgeStartNs)
+  })
+
+  it('should encode/decode to the same value when using the PathwayCodec, base64 and the deprecated ctx key', () => {
+    const ctx = {
+      pathwayStartNs: 1685673482722000000,
+      edgeStartNs: 1685673506404000000
+    }
+    const carrier = {}
+    ctx.hash = computePathwayHash('test-service', 'test-env',
+      ['direction:in', 'group:group1', 'topic:topic1', 'type:kafka'], Buffer.from('0000000000000000', 'hex'))
+
+    DsmPathwayCodec.encode(ctx, carrier)
+    carrier['dd-pathway-ctx'] = carrier['dd-pathway-ctx-base64']
+    delete carrier['dd-pathway-ctx-base64']
+    const decodedCtx = DsmPathwayCodec.decode(carrier)
+
+    expect(decodedCtx.hash.toString()).to.equal(ctx.hash.toString())
+    expect(decodedCtx.pathwayStartNs).to.equal(ctx.pathwayStartNs)
+    expect(decodedCtx.edgeStartNs).to.equal(ctx.edgeStartNs)
+  })
+
+  it('should encode/decode to the same value when using the PathwayCodec and the deprecated encoding', () => {
```

Comment:
what do you think about adding a test with a hard coded byte64 encoded `dd-pathway-ctx-base64`?

Reply:
I've added hardcoded versions for the final two tests